### ISSUE TITLE
Clearing data to develop

### ISSUE THREAD

**Issue by craftzneko** [grafana/alloy#1184]
### What's wrong?

I am trying alloy and loki so using it in a development environment. I have made some config changes and want to have alloy reasses all the existing log files and send them to loki again. My loki config looks like this

```yaml
auth_enabled: false

server:
  http_listen_port: 3100

common:
  ring:
    instance_addr: 127.0.0.1
    kvstore:
      store: inmemory
  replication_factor: 1
  path_prefix: /tmp/loki

schema_config:
  configs:
  - from: 2020-05-15
    store: tsdb
    object_store: filesystem
    schema: v13
    index:
      prefix: index_
      period: 24h

storage_config:
  filesystem:
    directory: /tmp/loki/chunks
```
My alloy config looks like this
```yaml
local.file_match "local_files" {
    path_targets = [{"__path__" = "/test/logs/*.log"}]
    sync_period = "5s"
}

loki.source.file "log_scrape" {
   targets    = local.file_match.local_files.targets
   forward_to = [loki.process.filter_logs.receiver]
   tail_from_end = true
}

loki.process "filter_logs" {
  stage.drop {
       source = ""
       expression  = ".*Connection closed by authenticating user root"
       drop_counter_reason = "noisy"
    }
  forward_to = [loki.process.add_static_label.receiver]
  }

loki.process "add_static_label" {
    forward_to = [loki.write.grafana_loki.receiver]

  stage.static_labels {
      values = {
        logs = "expressapi",
        env = "production",
      }
  }
}

  loki.write "grafana_loki" {
  endpoint {
    url = "http://lokitest.cardiff.gov.uk:3100/loki/api/v1/push"

    // basic_auth {
    //  username = "admin"
    //  password = "admin"
    // }
  }
}
```

I stopped loki service on the server and alloy on the client, cleared the c:\tmp\loki folder on server and C:\ProgramData\GrafanaLabs\Alloy\data on client, rebooted both, restarted loki service and alloy service. So as far as i can see this should now mean its clean ready to be run again. I found no advice in any of the documentation about this

Logs are not ingested and appears as if loki is empty with no labels. Alloy says its healthy and if i add a new log file it has never seen before it seems to ingest it but doesnt process any existing logs. So where is it storing this data so i can clear in between testing!!!

### Steps to reproduce

as above

**Comment by rfratto** [comment]
> Logs are not ingested and appears as if loki is empty with no labels. Alloy says its healthy and if i add a new log file it has never seen before it seems to ingest it but doesnt process any existing logs. So where is it storing this data so i can clear in between testing!!!

Hi, you're likely experiencing the behaviour of [loki.source.file's "positions" file](https://grafana.com/docs/alloy/latest/reference/components/loki.source.file/#component-behavior). The positions file is stored on disk and tells the component byte offsets for where tailing should resume if Alloy were to restart. 

The positions file is stored in Alloy's data directory, controlled by the `--storage.path` argument when running Alloy. If you're running on Linux using our system packages, we set `--storage.path` to `/var/lib/alloy/data`. You should see a subfolder in there called `loki.source.file.log_scrape` where the positions file will be found. You can delete that file before starting Alloy, or if you don't want to retain anything in between runs, you can delete the entire storage directory before starting Alloy.

**Comment by craftzneko** [comment]
I repeated the process again with debug logs on. So c:\programdata\grafanalabs\alloy\data cleared and service restarted.

I see logs like this

ts=2024-07-02T17:55:49.6273866Z level=info msg="Seeked C:\\test\\logs\\XpressRegisterChecker_20240702.log - &{Offset:1143145 Whence:0}" component_path=/ component_id=loki.source.file.log_scrape
ts=2024-07-02T17:55:49.6273866Z level=info msg="tail routine: started" component_path=/ component_id=loki.source.file.log_scrape component=tailer path=C:\test\logs\XpressRegisterChecker_20240702.log

which if i am reading correctly it still using an offset for the file

**Comment by rfratto** [comment]
I'd like to make sure that `C:\ProgramData\GrafanaLabs\Alloy\data` is the correct storage path for your install. 

Can you navigate to `http://localhost:12345/debug/pprof/cmdline?debug=1` on the machine where Alloy is running? That will report the command line used to start Alloy and should show the value of `--storage.path` (if set).

**Comment by craftzneko** [comment]
Here is the result, seems to be the same path ![image](https://github.com/grafana/alloy/assets/662108/8aeb083d-eb0b-40e4-8fd7-ca01e9c9455e)

**Comment by rfratto** [comment]
Thanks, good to confirm that's where the positions file is being kept. 

Given you've set `tail_from_end = true` in your configuration (where all files will be tailed from the end of the file instead of the beginning), I don't think we can use those logs as an indication that it's a positions file issue, since it'll seek to the end of the file when starting up. 

Are these files actively being written to? Did you mean to set `tail_from_end`? 

Another issue could be on the Loki side, but you'd have to look for more log messages with `component_id=loki.write.grafana_loki` in the log line to see if there's anything relevant there.

**Comment by craftzneko** [comment]
Cheers that was actually the issue i didn't see that in my config, removed and now its pulling them all in
